dart.yaml

```diff
@@ -0,0 +1,137 @@
+package:
+  name: dart
+  version: 3.5.3
+  epoch: 0
+  description: The Dart SDK, including the VM, JS and Wasm compilers, analysis, core libraries, and more.
+  copyright:
+    - license: BSD-3-Clause
+  dependencies:
+    runtime:
+      - ca-certificates-bundle
+
+environment:
+  contents:
+    packages:
+      - bash
+      - bind-tools
+      - busybox
+      - ca-certificates
+      - ca-certificates-bundle
+      - curl
+      - git
+      - python3
+      - xz
+
+pipeline:
+  - name: "clone the source code"
+    runs: |
+      mkdir -p ${{targets.destdir}}/usr/lib/
+      mkdir -p ${{targets.destdir}}/runtime
+      git clone --depth=1 https://chromium.googlesource.com/chromium/tools/depot_tools.git /home/depot_tools
+      export PATH=$PATH:/home/depot_tools
+      mkdir -p /home/dart-sdk
+      cd /home/dart-sdk
+      fetch dart
+      cd sdk
+      git fetch --tags --all
+      git checkout tags/${{package.version}} -b ${{package.version}}
+      gclient sync
+
+  - if: ${{build.arch}} == "x86_64"
+    working-directory: /home/dart-sdk/sdk
+    runs: |
+      ./tools/build.py --mode release --arch x64 create_sdk
+      mv out/ReleaseX64/dart-sdk ${{targets.destdir}}/usr/lib/dart
+      cp /lib64/ld-linux-x86-64.so.2 ${{targets.destdir}}/runtime/
+
+  - if: ${{build.arch}} == "aarch64"
+    working-directory: /home/dart-sdk/sdk
+    runs: |
+      ./tools/build.py --mode release --arch arm64 create_sdk
+      mv out/ReleaseARM64/dart-sdk ${{targets.destdir}}/usr/lib/dart
+      cp /lib/ld-linux-aarch64.so.1 ${{targets.destdir}}/runtime/
+
+  - name: "symlink dart binary and copy required libraries to runtime dir"
+    runs: |
+      mkdir -p ${{targets.destdir}}/usr/bin
+      FILES="/etc/nsswitch.conf /lib/libc.so.6 /lib/libdl.so.2 /lib/libm.so.6 /lib/libnss_dns.so.2 /lib/libpthread.so.0 /lib/libresolv.so.2 /lib/librt.so.1"
+      for file in $FILES; do
+        cp -L $file ${{targets.destdir}}/runtime/
+      done
+      mkdir -p /runtime/etc/ssl
+      mkdir -p /runtime/usr/share
+      cp -R /etc/ssl/certs /runtime/etc/ssl
+      cp -R /usr/share/ca-certificates /runtime/usr/share
+      ln -sf /usr/lib/dart/bin/dart "${{targets.destdir}}"/usr/bin/dart
+
+  - uses: git-checkout
+    with:
+      repository: https://github.com/dart-lang/sdk
+      tag: ${{package.version}}
+      expected-commit: 179da3ba67ead156a3ba718bd798cb7a1728e741
+
+update:
+  enabled: true
+  ignore-regex-patterns:
+    - '.*\.dev$'
+    - '.*\.alpha$'
+    - '.*\.beta$'
+    - "analyzer-*"
+    - "meta-*"
+    - "merge*"
+  git: {}
+  schedule:
+    period: weekly
+    reason: upstream releases stable version every 3 months but polling can be done weekly so that we don't miss things.
+
+test:
+  environment:
+    contents:
+      packages:
+        - ca-certificates-bundle
+        - git
```

Comment:
I don't see direct calls to these in the test cases. it means dart itself calls it. Please add them as runtime dependencies of dart; otherwise one will have to repeat this again in the image building; any custom image that needs dart; and any other higher deployment that tries to depend on and use dart.

Reply:
Thanks, I was using `git` to clone samples repo from dart and run some tests but now I've converted those to use dart itself. `git` is not longer required. 
`ca-certificates-bundle` is still required because by default test-environment is still coming with `r0` as of now. I think we need digests update here after https://github.com/wolfi-dev/os/pull/29948 was merged to get `r1` propogated in test environment.